### DISCUSSION TITLE
Intermittent "Module parse failed" errors

### DISCUSSION THREAD

**Discussion by JasonGore** [webpack/webpack#17721] [General]
I work on a large scale project where webpack runs in CI hundreds of times per day. For the last few weeks, we are seeing an intermittent but persistent rate (~1%) of "Module parse failed" errors with unterminated strings, regular expressions, and unexpected input. In all cases, the parsed input doesn't match what's actually on disk. In some cases parsed input seems truncated while in others parsed input has appended content. Actual file contents on disk seem valid in all cases, and don't change between successful and unsuccessful runs.

This seems completely unrelated to obvious loader config issues that would lead to consistently reproducible failures.  It happens with completely different webpack loader configs and there seem to be no common loaders involved. I don't think it could be a broad issue with file I/O or handles as we run a wide variety of processes and build tasks and none fail with similar file I/O issues. As a result, this seems to be some sort of Node or webpack architectural issue. It seems like there is some type of pollution or corruption of global state or file streams that's causing these intermittent types of errors. I wanted to post here in case anyone has ever encountered anything remotely similar as it's been a real head scratcher for us.

We do have many bundles running in parallel on these VMs in CI, but they are running in their own webpack CLI-invoked Node processes and should be isolated from each other. I'm curious if anybody knows of any way that state could somehow pollute across bundles happening in separate processes.

I could also use any help trying to pinpoint the issue more closely, such as tracing loader chain activity, printing file contents/stats, tracing parsed input, etc.

webpack: 5.82.0, 5.88.2
Node: v18.17.1
OS: Ubuntu 22.04.2 LTS

```
Module parse failed: Unterminated regular expression (2:1)
File was processed with these loaders:
* ../../node_modules/source-map-loader/dist/cjs.js
* ../../node_modules/@griffel/webpack-extraction-plugin/src/webpackLoader.js
You may need an additional loader to handle the result of these loaders.
| export {};
> /
```

```
Module parse failed: Unterminated string constant (30:14)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| export * from "./createServiceContainer";
|
> export * from "./MarginProperty
```

**Comment by alexander-akait** [comment]
@JasonGore Sorry for delay, still valid?

**Comment by JasonGore** [comment]
We still see the issue occasionally in our CI but not as frequently as when I created this conversation item. I'd still be curious to see if there's any insight into if and how state could cross processes, whether via cross-process cache, file I/O, or other.

**Reply by alexander-akait** [reply]
How do you run webpack? Also check you don't have intersecting loader (i.e. check regexp in `test`/`include`/`exclude`). Also do you generate files before compilation and during compilation (out of process webpack itself)?
